esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -1399,6 +1399,53 @@ void EsdlServiceImpl::getTargetResponseFile(IEspContext & context,
         resp.loadFile(respfile);
 }
 
+
+void EsdlServiceImpl::processRoxieTestRequest(IEspContext &context, IEsdlDefService &srvdef, IEsdlDefMethod &mthdef, Owned<IPropertyTree> &tgtcfg, Owned<IPropertyTree> &tgtctx, StringBuffer &reqStr)
```

Comment:
This is a general comment about the entire change. Replicating this logic doesn't seem like the right solution. It would be better to encapsulate a single method that can take the original ESP request and return the complete request as it will be presented to the back end. This will allow all request processing, existing and future, to be implemented once and used wherever needed.

Keep in mind that the back end may be a roxie, but it may also be a Java or C++ plugin, and could be other things in the future. If this feature must assume a roxie back end, then an input parameter constraining the request to roxie backends would be OK.

Reply:
@timothyklemm Please review the refactored version

Reply:
@timothyklemm I've pushed another small change as recommended